activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -141,10 +141,18 @@ def create_all
         end
       end
 
-      def for_each
+      def setup_initial_database_yaml
         return {} unless defined?(Rails)
 
-        databases = Rails.application.config.load_database_yaml
+        begin
+          Rails.application.config.load_database_yaml
+        rescue
+          $stdout.puts "Rails couldn't infer whether you are using multiple databases from your database.yml. If you'd like to use this feature, please simplify your ERB."
```

Comment:
`$stderr` maybe?

I think for the message we could do:

"Rails couldn't infer whether you are using multiple databases from your database.yml and can't generate the test for the non-primary databases. If you'd like to use this feature, please simplify your ERB."

Reply:
Done.